src/coord/src/coord.rs

```diff
@@ -1379,15 +1379,6 @@ impl Coordinator {
                 return;
             }
 
-            // Cancel the peek. We use an `if let` because the peek could be completed
-            // and removed before the cancellation is received.
-            if let Some((channel, count)) = self.pending_peeks.get(&conn_id) {
-                for _ in 0..*count {
-                    channel
-                        .send(PeekResponse::Canceled)
-                        .expect("Peek channel closed prematurely");
-                }
-            }
```

Comment:
Everything else in the commit makes sense to me other than this change. It's possible it doesn't need to make sense to me, or that once we have the 1:1 contract with peek responses it isn't needed. But it does seem like this has the potential to do something different that I don't fully understand.

Reply:
This code was already broken, we just didn't know it. The intent of this code was to send num worker cancel messages to `message_worker()`. But that's not what it did. It actually just sent cancel messages to the `.fold` in `sequence_peek`, leaving the pending_peeks open. Removing this block is correct.

What we could add here is explicitly dropping the pending_peeks entry, so we do not need to wait for dataflow workers to respond.

Reply:
In Slack, Frank said:

> awaiting exactly one message from each worker makes it a lot easier to avoid other bugs, like a cancelled peek followed by a new peek, and us not knowing which peek a Cancelled response refers to

So the proposed design is to await those cancellation messages. I think it's safer to leave this contract in place (i.e. one message per worker per peek) and optimize performance later iff it becomes an issue.